custodian/vasp/handlers.py

```diff
@@ -935,62 +934,46 @@ def __init__(self, wall_time=None, buffer_time=300,
                 Should be used with LWAVE = .True. to be useful. If this is
                 True, the STOPCAR is written with LABORT = .TRUE. instead of
                 LSTOP = .TRUE.
-            auto_continue (bool): Use the auto-continue functionality within the
-                VaspJob by ensuring Vasp doesn't delete the STOPCAR
         """
         if wall_time is not None:
             self.wall_time = wall_time
         elif "PBS_WALLTIME" in os.environ:
             self.wall_time = int(os.environ["PBS_WALLTIME"])
         elif "SBATCH_TIMELIMIT" in os.environ:
-            self.walltime = int(os.environ["SBATCH_TIMELIMIT"]) 
+            self.wall_time = int(os.environ["SBATCH_TIMELIMIT"]) 
         else:
             self.wall_time = None
         self.buffer_time = buffer_time
-        self.start_time = datetime.datetime.now()
+        if "JOB_START_TIME" in os.environ:
+            # Enables use of setting JOB_START_TIME as the start
+            # time to use for the walltime handler.  The easiest
+            # way to use this is by executing export JOB_START_TIME=`date`
+            # in the environment (e. g. in a queue submission script)
+            self.start_time = datetime.datetime.strptime(
+                os.environ["JOB_START_TIME"], "%a %b %d %H:%M:%S %Z %Y")
+        else:
+            self.start_time = datetime.datetime.now()
         self.electronic_step_stop = electronic_step_stop
         self.electronic_steps_timings = [0]
         self.prev_check_time = self.start_time
-        self.prev_check_nscf_steps = 0
-        self.auto_continue = auto_continue
 
     def check(self):
         if self.wall_time:
             run_time = datetime.datetime.now() - self.start_time
             total_secs = run_time.total_seconds()
+            outcar = Outcar("OUTCAR")
             if not self.electronic_step_stop:
-                try:
-                    # Intelligently determine time per ionic step.
-                    o = Oszicar("OSZICAR")
-                    nsteps = len(o.ionic_steps)
-                    time_per_step = total_secs / nsteps
-                except Exception:
-                    time_per_step = 0
+                # Determine max time per ionic step.
+                outcar.read_pattern({"timings": "LOOP\+.+real time(.+)"},
+                                postprocess=float)
+                time_per_step = np.max(outcar.data.get('timings')) or 0
```

Comment:
They seem to be pretty accurate in my experience if you use "real time".  "CPU time" is all over the place in my experience, and I'm not really sure why.